### ISSUE TITLE
Deprecate and remove Enum-like objects

### ISSUE THREAD

**Issue by blaine-arcjet** [arcjet/arcjet-js#1833]
The initial design of the SDK used TypeScript enums for various configuration. We've removed those in favor of string unions, which fit more naturally with the way people write both TypeScript and JavaScript. However, we kept around objects that behaved like the enums for migration. Our docs haven't recommended those for a long time, and we're moving towards string values that translate directly to Wasm via jco.

We should deprecate and remove these.

**Comment by e-moran** [comment]
We decided in the team call today that we should target deprecating these for the beta, and remove them at a later stage.

The deprecation should will print a warn-level message advising users that they should consider switching to string unions soon. We will then remove them in a future update. The deprecation is tracked by #2621.

**Comment by e-moran** [comment]
From looking in to this, it seems that the objects we used to use were generated by protobuf. I'm not sure what we actually want to remove here, since they are still used for rpc.

**Comment by blaine-arcjet** [comment]
> From looking in to this, it seems that the objects we used to use were generated by protobuf. I'm not sure what we actually want to remove here, since they are still used for rpc.

This is incorrect, the Enum-like objects are hand-written at https://github.com/arcjet/arcjet-js/blob/main/protocol/index.ts
For example, https://github.com/arcjet/arcjet-js/blob/main/protocol/index.ts#L15-L18 

These all use [ArcjetEnum](https://github.com/arcjet/arcjet-js/blob/main/protocol/index.ts#L12)

**Comment by davidmytton** [comment]
These were deprecated in https://github.com/arcjet/arcjet-js/pull/2684 and will be removed in the 1.0.0 release.
